bugbot/rules/not_landed.py

```diff
@@ -232,6 +232,20 @@ def attachment_handler(attachments, data):
 
             if "phab" in res:
                 if res["phab"]:
+                    # Check for stackGraph dependencies
+                    rev = PHAB_URL_PAT.search(
+                        base64.b64decode(attachment["data"]).decode("utf-8")
+                    ).group(1)
+                    try:
+                        revision_data = self.phab.load_revision(rev_id=int(rev))
+                        stack_graph = revision_data.get("fields", {}).get(
+                            "stackGraph", {}
+                        )
+                        current_revision_phid = revision_data.get("phid")
+                        dependencies = stack_graph.get(current_revision_phid, [])
+                        data[bugid]["dependencies"] = dependencies
+                    except PhabricatorRevisionNotFoundException:
+                        pass
```

Comment:
To improve clarity, it's advisable to relocate this logic into its own function instead of enclosing it in a try-except block. Subsequently, you can confine the try-except to the `load_revision()` method invocation.

Reply:
Done in 133e960a45499c75452919669cfa668c3711c86d!

Reply:
The try-except block should be limited to the logic it is meant to handle, specifically the load_revision() method invocation.

Reply:
Done in 1b41a36d077db1d0e2b12746330915733d9bb65f!